cpp/src/arrow/compute/kernels/hash_aggregate.cc

```diff
@@ -2939,26 +3287,26 @@ Result<Datum> GroupBy(const std::vector<Datum>& arguments, const std::vector<Dat
     ARROW_ASSIGN_OR_RAISE(Datum transposition, groupers[0]->Consume(other_keys));
     groupers[thread_index].reset();
 
-    for (size_t i = 0; i < kernels.size(); ++i) {
+    for (size_t idx = 0; idx < kernels.size(); ++idx) {
```

Comment:
Just a nit: but my clang-tidy kept showing me that this `Declaration shadows a local variable`. Will revert it if not needed.